src/sqlfluff/utils/reflow/elements.py

```diff
@@ -0,0 +1,606 @@
+        # NOTE: Reverse slice
+        parent_segment = None
+        for ps in root_segment.path_to(next_seg)[::-1]:
+            if ps.segment.is_type(align_within):
+                parent_segment = ps.segment
+            if ps.segment.is_type(align_boundary):
+                break
+
+        if not parent_segment:
+            reflow_logger.debug(
+                "    No Parent found for alignment case. Treat as single."
+            )
+            return " "
+
+        # We've got a parent. Find some siblings.
+        reflow_logger.debug("    Determining alignment within: %s", parent_segment)
+        siblings = []
+        for sibling in parent_segment.recursive_crawl(segment_type):
+            # Purge any siblings with a boundary between them
+            if not any(
+                ps.segment.is_type(align_boundary)
+                for ps in parent_segment.path_to(sibling)
+            ):
+                siblings.append(sibling)
+            else:
+                reflow_logger.debug(
+                    "    Purging a sibling because they're blocked "
+                    "by a boundary: %s",
+                    sibling,
+                )
+
+        # Is the current indent the only one on the line?
+        if any(
+            # Same line
+            sibling.pos_marker.working_line_no == next_seg.pos_marker.working_line_no
+            # And not same position (i.e. not self)
+            and sibling.pos_marker.working_line_pos
+            != next_seg.pos_marker.working_line_pos
+            for sibling in siblings
+        ):
+            reflow_logger.debug("    Found sibling on same line. Treat as single")
+            return " "
+
+        # Work out the current spacing before each.
+        last_code = None
+        max_desired_line_pos = 0
+        for seg in parent_segment.raw_segments:
+            for sibling in siblings:
+                # NOTE: We're asserting that there must have been
+                # a last_code. Otherwise this won't work.
+                if (
+                    seg.pos_marker.working_loc == sibling.pos_marker.working_loc
+                    and last_code
+                ):
+                    loc = last_code.pos_marker.working_loc_after(last_code.raw)
+                    reflow_logger.debug(
+                        "    loc for %s: %s from %s",
+                        sibling,
+                        loc,
+                        last_code,
+                    )
+                    if loc[1] > max_desired_line_pos:
+                        max_desired_line_pos = loc[1]
+            if seg.is_code:
+                last_code = seg
+
+        desired_space = " " * (
+            1 + max_desired_line_pos - whitespace_seg.pos_marker.working_line_pos
+        )
+        reflow_logger.debug(
+            "    desired_space: %r (based on max line pos of %s)",
+            desired_space,
+            max_desired_line_pos,
+        )
+        return desired_space
```

Comment:
This looks like the main chunk of code for identifying `align_aliases` violations

 (and then  `_handle_respace__inline_with_space` applies the fixes)

I dont have any concerns with this code.. 


This seems more reliable than the PR i put together, because you're actually checking `is_code()` which is an extra check to ensure `this segment has no subsegments`

My only concern is that the new default configs arent mentioned in `rules/config_info.py` ([example](https://github.com/sqlfluff/sqlfluff/pull/3762/files#diff-1fdcdfc3c128b96b64ddafed38baa65954da0cb213e7d72833307e5d7d9faaa5R37))

Reply:
I saw the `Reflow README` mentioned a new `layout` type of config, so wasnt sure if there was some default configuration you were doing somewhere else

Reply:
Yeah - I think we totally need better documentation for this. I've got that as a separate task for the master issue (#3673).

Thanks for the review!